docs/zenith/developer/guides.md

```diff
@@ -0,0 +1,17 @@
+---
+slug: /zenith/developer/guides
+displayed_sidebar: "zenith"
+---
+
+import GuideIndex from '@site/src/components/molecules/guideIndex.js'
+
+# Guides
+
+{@include: ../partials/_experimental.md}
+
+Learn more about Dagger programming using the step-by-step walkthroughs and guides in this section.
+
+- [Advanced Module Development](./guides/191108-advanced-module-development.md)
+- [Publish Modules to the Daggerverse](./guides/821742-publish-modules.md)
+
+<GuideIndex />
```

Comment:
@crjm I would like to reuse this component for Zenith guides. Can we modify it to accept a `dir` argument, which is the directory it should look up for the guides? Currently it is hardcoded to the `current` version guides, so does not display `zenith` version guides.

Reply:
Essentially, if configured correctly, it should display these guides in the index
- Advanced Module Development
- Publish Modules to the Daggerverse

Reply:
Pushed support for zenith guides to the index.
LMK if you think something is looking odd!

BTW, I only added the support for the Zenith guides, left the content of the index page to you (right now there's some duplication with the guide list on top of the index).

Reply:
Thanks @crjm! I have adjusted the content of the guides index page. It looks good.

Question: Currently, we have only one Guides index page, for the "programming Dagger" section. There is a possibility of adding another Guides index page, in the "using Dagger" section, listing a different set of guides. How could I do this with the GuidesIndex component?

Reply:
So, right now we are writing a JSON file in `/website/static/guides.json` with a custom guides plugin being declared in `docusaurus.config.js`. 
This file stores the parsed guides so we can consume them in the `GuidesIndex` component. 

In order to make the plugin aware of a new guides folder, we need to provide the folder to the plugin. To go faster, I didn't make it smart, but I duplicated code for the zenith and current guides:
https://github.com/dagger/dagger/pull/6015/files#diff-d2690b736219c540a0ae0ef4795185abf126c55fb0280e3ed5a8a1826d0391d7R47

In case we want to add another source of guides, we should pass the path to the plugin trough the `docusaurus.config.js`, resolve the dir inside the plugin, and make the guides available in a new JSON field as the output of the plugin. This way, you should be able to consume the data in the JSON from the component itself.

Overall, I would try to make the implementation smarter so it can get an array of directories and that's it, so you don't have to make all the changes above, but LMK! Happy to sync to discuss this further too.

Reply:
I think it makes sense to add this improvement, because I think we will shortly want to have both user and developer guides under separate nodes of the nav tree. We can simplify the file resolution by assuming for the moment that the guides to be listed will always be in a `guides` subdir of the directory containing the index file. So for example:

```
// file is current/guides.mdx -> display index for current/guides/*
// file is zenith/user/guides.mdx -> display index for zenith/user/guides/*
// etc
```

Reply:
We discussed this and it is not required atm. No action needed.